fetch.bs

```diff
@@ -5630,6 +5622,82 @@ if the script checks that the URL has the right hostname.
 
 
 
+<h2 id=data-urls><code>data:</code> URLs</h2>
+
+<p>For an informative description of <code>data:</code> URLs, see RFC 2397. This section replaces
+that RFC's normative processing requirements. [[RFC2397]]
+
+<p>The <dfn export><code>data:</code> URL processor</dfn> takes a <a for=/>URL</a>
+<var>dataURL</var> and then runs these steps:
+
+<ol>
+ <li><p>Assert: <var>dataURL</var>'s <a for=URL>scheme</a> is "<code>data</code>".
+
+ <li><p>Let <var>input</var> be the result of running the <a>URL serializer</a> on
+ <var>dataURL</var> with the <i>exclude fragment flag</i> set.
+
+ <li><p>Remove the leading "<code>data:</code>" string from <var>input</var>.
+
+ <li><p>Let <var>position</var> point at the start of <var>input</var>.
+
+ <li><p>Let <var>encodedMimeType</var> be the result of
+ <a>collecting a sequence of code points</a> that are not equal to U+002C (,), given
+ <var>position</var>.
+
+ <li><p>If <var>position</var> is past the end of <var>input</var>, then return failure.
+
+ <li><p>Advance <var>position</var> by 1.
+
+ <li><p>Let <var>encodedBody</var> be the remainder of <var>input</var>.
+
+ <li><p>Let <var>mimeTypeBytes</var> be the <a>string percent decoding</a> of
+ <var>encodedMimeType</var>.
+ <!-- Note: implementations leave the percent-encoded bits around. That strikes me as rather broken,
```

Comment:
In other words, implementations don't percent decode the mime type at all? Are any of the tests failing everywhere because of this? If so, going with the broken reality seems preferable.

Reply:
Really, nobody wants to put effort in this? I guess I should check again to make sure no implementation does it anywhere, but it seems rather sad and it means you cannot have commas anywhere (although the case for them is rather theoretical I suppose).

Reply:
Having no context on this, I guess I don't see why percent decoding is the default reasonable thing to do. Is it just generally true of "parts of URLs"? This part is constrained by the syntax of mime types, does that even allow for anything that needs to be percent encoded?

Reply:
Well, that's why you'd percent-decode before you try to parse the MIME type. And yeah, generally in URLs "a" and "%61" would be equivalent and consumers of URLs are expected to treat those equivalently.

Reply:
So, a cast that isn't entirely far fetched is `data:video/ogg;codecs=theora,vorbis,...`. Even though I don't know of a case where the codecs are *needed*, it is valid to have a comma in a media type.

No objections to try to spec the more reasonable behavior, but it'd be good if it's isolated to a few tests, so that if they're all still failing in a year or so, reversing course will be a small effort.